kevlar/include/kevlar_bits/model/binomial/berry_inla.hpp

```diff
@@ -0,0 +1,372 @@
+#pragma once
+#include <Eigen/Dense>
+#include <algorithm>
+#include <kevlar_bits/model/base.hpp>
+#include <kevlar_bits/model/binomial/common/fixed_n_default.hpp>
+#include <kevlar_bits/model/fixed_single_arm_size.hpp>
+#include <kevlar_bits/util/algorithm.hpp>
+#include <kevlar_bits/util/legendre.hpp>
+#include <kevlar_bits/util/macros.hpp>
+#include <kevlar_bits/util/math.hpp>
+#include <kevlar_bits/util/types.hpp>
+
+namespace kevlar {
+namespace model {
+namespace binomial {
+
+template <class ValueType, int ARMS>
+struct BerryINLA : FixedSingleArmSize, ModelBase<ValueType> {
```

Comment:
Most of the code here is copied/mildly changed from the DirectBayes implementation and from the `fast_inla_ext.cpp` implementation.